contrib/executor/zap/pkg/runner/runner.go

```diff
@@ -0,0 +1,229 @@
+	ZapHome string
+	Scraper scraper.Scraper
+}
+
+var _ runner.Runner = &ZapRunner{}
+
+// NewRunner creates a new ZapRunner
+func NewRunner(ctx context.Context, params envs.Params) (*ZapRunner, error) {
+	output.PrintLogf("%s Preparing test runner", ui.IconTruck)
+
+	var err error
+	r := &ZapRunner{
+		Params:  params,
+		ZapHome: os.Getenv("ZAP_HOME"),
+	}
+
+	output.PrintLogf("%s Preparing scraper", ui.IconTruck)
+	r.Scraper, err = factory.TryGetScrapper(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Run executes the test and returns the test results
+func (r *ZapRunner) Run(ctx context.Context, execution testkube.Execution) (result testkube.ExecutionResult, err error) {
+	if r.Scraper != nil {
+		defer r.Scraper.Close()
+	}
+	output.PrintLogf("%s Preparing for test run", ui.IconTruck)
+
+	testFile, workingDir, err := content.GetPathAndWorkingDir(execution.Content, r.Params.DataDir)
+	if err != nil {
+		output.PrintLogf("%s Failed to resolve absolute directory for %s, using the path directly", ui.IconWarning, r.Params.DataDir)
+	}
+
+	fileInfo, err := os.Stat(testFile)
+	if err != nil {
+		return result, err
+	}
+
+	var zapConfig string
+
+	if fileInfo.IsDir() {
+		// assume the ZAP config YAML has been passed as test argument
+		zapConfig = filepath.Join(workingDir, execution.Args[len(execution.Args)-1])
+	} else {
+		// use the given file config as ZAP config YAML
+		workingDir = r.Params.DataDir
+		zapConfig = testFile
+	}
+
+	output.PrintLogf("%s Reading options", ui.IconWorld)
+	options := Options{}
+	err = options.UnmarshalYAML(zapConfig)
+	if err != nil {
+		return *result.WithErrors(err), nil
+	}
+
+	// determine the actual ZAP script and args to run
+	output.PrintLogf("%s Processing test type", ui.IconWorld)
+	scanType := strings.Split(execution.TestType, "/")[1]
+	reportFolder := filepath.Join(r.Params.DataDir, "reports")
+	err = os.Mkdir(reportFolder, 0700)
+	if err != nil {
+		return *result.WithErrors(err), nil
+	}
+	output.PrintLogf("%s Preparing reports folder", ui.IconFile)
+	reportFile := filepath.Join(reportFolder, fmt.Sprintf("%s-report.html", execution.TestName))
+	scriptName := zapScript(scanType)
+	args := zapArgs(scanType, options, reportFile)
+
+	output.PrintLogf("%s Preparing variables", ui.IconWorld)
+	envManager := env.NewManagerWithVars(execution.Variables)
+	envManager.GetReferenceVars(envManager.Variables)
+	// simply set the ENVs to use during execution
+	for _, env := range execution.Variables {
+		os.Setenv(env.Name, env.Value)
+	}
+
+	// convert executor env variables to runner env variables
+	for key, value := range execution.Envs {
+		os.Setenv(key, value)
+	}
+
+	// when using file based ZAP parameters it expects a /zap/wrk directory
+	// we simply symlink the directory
+	os.Symlink(workingDir, filepath.Join(r.ZapHome, "wrk"))
+
+	output.PrintLogf("%s Running ZAP test", ui.IconMicroscope)
+	logs, err := executor.Run(r.ZapHome, scriptName, envManager, args...)
```

Comment:
ideally, we need to prepare a command call from command and arguments defined in executor spec. you can check other executors. Sometimes, this is just direct parameters, in many cases, it's a template param, that we replace by actual value in executor

Reply:
Need to print command and args

Reply:
added